SparseGrids/tsgCoreOneDimensional.cpp

```diff
@@ -392,6 +393,72 @@ const char* OneDimensionalMeta::getHumanString(TypeOneDRule rule){
     }
 }
 
+// Generate the n roots of the n-th degree orthogonal polynomial.
+// ref_points and ref_points are used in the computation of the integrals that form the elements of the Jacobi matrix.
+double poly_eval(int nroots, std::vector<double> roots, double x) {
+    double eval_value = 1.0;
+    for (int i=0; i<nroots; i++) {
+        eval_value *= (x - roots[i]);
+    }
+    return(eval_value);
+}
+std::vector<std::vector<double>> OneDimensionalOrthPolynomials::getRootCache(
+        int n,
+        std::function<double(double)> &weight_fn,
+        const std::vector<double> &ref_weights,
+        const std::vector<double> &ref_points) {
+
+    // Initialize and compute the integral weights corresponding to weight_fn.
+    std::vector<std::vector<double>> root_cache(n);
+    std::vector<double> roots;
+    #ifdef Tasmanian_ENABLE_BLAS
+    if (ref_points.size() != ref_weights.size()) {
+        throw std::invalid_argument("The number of reference points does not match the point of reference weights!\n");
+    }
+    int nref = ref_points.size();
+    std::vector<double> integral_weights(nref);
+    for (int j=0; j<nref; j++) {
+        integral_weights[j] = weight_fn(ref_points[j]) * ref_weights[j];
+    }
+    // Compute the roots incrementally.
+    std::vector<double> poly_m1_vals(nref, 0.0), poly_vals(nref, 1.0);
+    std::vector<double> alpha(n), beta(n-1), offdiag(n-1);
+    double alpha_numr, alpha_denm, beta_numr, beta_denm;
+    for (int i=0; i<n; i++) {
+        // Form the tridiagonal vectors of the Jacobi matrix.
+        alpha_numr = 0.0;
+        alpha_denm = 0.0;
+        beta_numr = 0.0;
+        beta_denm = 0.0;
+        for (int j=0; j<nref; j++) {
+            alpha_numr += ref_points[j] * pow(poly_vals[j], 2) * integral_weights[j];
```

Comment:
* use `std::pow()` as opposed to just `pow` which can be in any namespace
* the integer version of `std::pow()` has been deprecated, use `std::pow(..., 2.0)`
    * the compiler will convert 2 to 2.0, but it's better to be explicit
* don't use `std::pow()` at all here, you can just use `poly_vals[j] * poly_vals[j]`
    * I'm worried that `pow()` will use logarithms for potentially non-integer powers which will waste many flops

Reply:
Changed. ``pow`` is no longer used.